modules/lvgl/input/lvgl_pointer_input.c

```diff
@@ -21,74 +21,77 @@ struct lvgl_pointer_input_config {
 	bool invert_y;
 };
 
+struct lvgl_pointer_input_data {
+	struct lvgl_common_input_data common_data;
+	uint32_t point_x;
+	uint32_t point_y;
+};
+
 static void lvgl_pointer_process_event(const struct device *dev, struct input_event *evt)
 {
 	const struct lvgl_pointer_input_config *cfg = dev->config;
-	struct lvgl_common_input_data *data = dev->data;
+	struct lvgl_pointer_input_data *data = dev->data;
 	lv_disp_t *disp = lv_disp_get_default();
 	struct lvgl_disp_data *disp_data = disp->driver->user_data;
 	struct display_capabilities *cap = &disp_data->cap;
-	lv_point_t *point = &data->pending_event.point;
+	lv_point_t *point = &data->common_data.pending_event.point;
 
 	switch (evt->code) {
 	case INPUT_ABS_X:
-		point->x = evt->value;
+		if (cfg->swap_xy) {
+			data->point_y = evt->value;
+		} else {
+			data->point_x = evt->value;
+		}
 		break;
 	case INPUT_ABS_Y:
-		point->y = evt->value;
+		if (cfg->swap_xy) {
+			data->point_x = evt->value;
+		} else {
+			data->point_y = evt->value;
+		}
 		break;
 	case INPUT_BTN_TOUCH:
-		data->pending_event.state = evt->value ? LV_INDEV_STATE_PR : LV_INDEV_STATE_REL;
+		data->common_data.pending_event.state =
+			evt->value ? LV_INDEV_STATE_PR : LV_INDEV_STATE_REL;
 		break;
 	}
 
 	if (!evt->sync) {
 		return;
 	}
 
-	/* adjust coordinates */
-	if (cfg->swap_xy) {
-		lv_coord_t tmp;
-
-		tmp = point->x;
-		point->x = point->y;
-		point->y = tmp;
-	}
+	point->x = data->point_x;
+	point->y = data->point_y;
 
 	if (cfg->invert_x) {
 		if (cap->current_orientation == DISPLAY_ORIENTATION_NORMAL ||
 		    cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_180) {
-			point->x = cap->x_resolution - point->x;
+			point->x = cap->x_resolution - data->point_x;
 		} else {
-			point->x = cap->y_resolution - point->x;
+			point->x = cap->y_resolution - data->point_x;
 		}
 	}
 
 	if (cfg->invert_y) {
 		if (cap->current_orientation == DISPLAY_ORIENTATION_NORMAL ||
 		    cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_180) {
-			point->y = cap->y_resolution - point->y;
+			point->y = cap->y_resolution - data->point_y;
 		} else {
-			point->y = cap->x_resolution - point->y;
+			point->y = cap->x_resolution - data->point_y;
 		}
 	}
 
 	/* rotate touch point to match display rotation */
 	if (cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_90) {
-		lv_coord_t tmp;
-
-		tmp = point->x;
-		point->x = point->y;
-		point->y = cap->y_resolution - tmp;
+		point->x = data->point_y;
```

Comment:
This doesn't look right. If `cfg->invert_x` or `cfg->invert_y` are set, AND the display orientation is rotated, you overwrite the inversion.

Reply:
Maybe this is what breaks input for me too.  I am using the Adafruit 2.8 shield and trying to just do a simple tab view.  The input seems to "work" by pressing the inverse x & y positions (like the input detection is rotated 180 degrees).